listenbrainz/spark/spark_reader.py

```diff
@@ -0,0 +1,85 @@
+import json
+import logging
+import pika
+import time
+import ujson
+
+from flask import current_app
+from listenbrainz import utils
+from listenbrainz.db import user as db_user, stats as db_stats
+from listenbrainz.webserver import create_app
+from listenbrainz.db.exceptions import DatabaseException
+from listenbrainz import config
+import sqlalchemy
+
+class SparkReader:
+    def __init__(self):
+        self.app = create_app() # creating a flask app for config values and logging to Sentry
+
+
+    def init_rabbitmq_connection(self):
+        """ Initializes the connection to RabbitMQ.
+
+        Note: this is a blocking function which keeps retrying if it fails
+        to connect to RabbitMQ
+        """
+        self.connection = utils.connect_to_rabbitmq(
+            username=current_app.config['RABBITMQ_USERNAME'],
+            password=current_app.config['RABBITMQ_PASSWORD'],
+            host=current_app.config['RABBITMQ_HOST'],
+            port=current_app.config['RABBITMQ_PORT'],
+            virtual_host=current_app.config['RABBITMQ_VHOST'],
+            error_logger=current_app.logger.error,
+        )
+
+
+    def callback(self, ch, method, properties, body):
+        """ Handle the data received from the queue and works accordingly.
```

Comment:
This docstring could be better. Handle the stats received from the queue and insert into the database accordingly.